src/randoop/operation/NonreceiverTerm.java

```diff
@@ -118,34 +123,24 @@ public ExecutionOutcome execute(Object[] statementInput, PrintStream out) {
     return Collections.emptyList();
   }
 
-  public void appendCode(Variable newVar, List<Variable> inputVars, StringBuilder b) {
-
-    if (type.isPrimitive()) {
-
-      b.append(PrimitiveTypes.boxedType(type).getName());
-      b.append(" ");
-      b.append(newVar.getName());
-      b.append(" = new ");
-      b.append(PrimitiveTypes.boxedType(type).getName());
-      b.append("(");
-      b.append(PrimitiveTypes.toCodeString(getValue()));
-      b.append(");");
-      b.append(Globals.lineSep);
-
-    } else {
-      b.append(Reflection.getCompilableName(type));
-      b.append(" ");
-      b.append(newVar.getName());
-      b.append(" = ");
-      b.append(PrimitiveTypes.toCodeString(getValue()));
-      b.append(";");
-      b.append(Globals.lineSep);
-    }
+  /**
+   * appendCode adds the string representation of the Nonreceiver value
+   * to the StringBuilder.
+   * Note: do not explicitly box primitive values.
```

Comment:
I moved the generation of the code for initialization and assignment into the Statement class, so the only remaining part of the method is the call to PrimitiveTypes.toCodeString(-).